### PR TITLE
PlanRepositoryTrigger

### PR DESCRIPTION
**Pull request by jstvz**
This PR adds the `PlanRepositoryTrigger` (PRT) model and associated handler for the `build_complete` signal. The handler looks up any PRTs for the completed build's `planrepo` and calls each instance's `fire()` method to enqueue the target build.

### CODE REVIEW

File: metaci/plan/models.py [hunk 1]
```diff
@@ -205,3 +206,49 @@ def get_absolute_url(self):
...
+    class Meta:
+        ordering = ["target_plan_repo", "source_plan_repo"]
+        unique_together = ("target_plan_repo", "source_plan_repo")
+        verbose_name_plural = "Plan Repository Triggers"
+        permissions = (
+            ("run_plan", "Run Plan"),
+            ("view_builds", "View Builds"),
+            ("rebuild_builds", "Rebuild Builds"),
+            ("org_login", "Login to Org"),
+        )
```
Comment (davisagli) [inline metaci/plan/models.py]: Are these permissions actually relevant here?

File: metaci/plan/models.py [hunk 2]
```diff
@@ -205,3 +206,49 @@ def get_absolute_url(self):
...
+        PlanRepository, on_delete=models.CASCADE, related_name="Triggers"
+    )
+    branch = models.CharField(max_length=255)
+    active = models.BooleanField(default=True)
+
+    objects = PlanRepositoryTriggerQuerySet.as_manager()
+
+    class Meta:
+        ordering = ["target_plan_repo", "source_plan_repo"]
+        unique_together = ("target_plan_repo", "source_plan_repo")
```
Comment (davisagli) [inline metaci/plan/models.py]: Hmm. What if we want to trigger builds for multiple branches in the target repository?

File: metaci/plan/handlers.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from django.dispatch import receiver
+from metaci.build.signals import build_complete
+from metaci.plan.models import PlanRepositoryTrigger
+
+
+@receiver(build_complete)
+def trigger_dependent_builds(sender, **kwargs):
+    build = kwargs.get("build")
+    status = kwargs.get("status")
```
Comment (davisagli) [inline metaci/plan/handlers.py]: Any reason not to just make the function signature be `(sender, build, status)`?

File: metaci/plan/models.py [hunk 4]
```diff
@@ -205,3 +206,49 @@ def get_absolute_url(self):
...
+            repo=self.target_plan_repo.repo, name=self.branch
+        )
+        return branch
+
+    def fire(self, finished_build):
+        build = Build(
+            repo=self.target_plan_repo.repo,
+            plan=self.target_plan_repo.plan,
+            planrepo=self.target_plan_repo,
+            commit=self.branch,
```
Comment (davisagli) [inline metaci/plan/models.py]: If the new build doesn't start running immediately, do we want it to use the state of this branch from the time the build was queued or the time it runs? If the former, we should resolve this to an actual commit sha like the plan form does in metaci/plan/forms.py

File: metaci/plan/models.py (hunk 1 above)
Comment (jstvz) [reply metaci/plan/models.py]: Cargo-culted a bit here. Will remove.

File: metaci/plan/models.py (hunk 2 above)
Comment (jstvz) [reply metaci/plan/models.py]: Good question. I hadn't considered the use case. I can add `branch` to `unique_together`.

File: metaci/plan/handlers.py (hunk 3 above)
Comment (jstvz) [reply metaci/plan/handlers.py]: My understanding is that it's less tightly coupled to the `build_complete` signal's arguments. ([docs](https://docs.djangoproject.com/en/2.1/topics/signals/#receiver-functions))

File: metaci/plan/models.py (hunk 4 above)
Comment (jstvz) [reply metaci/plan/models.py]: I was operating under the assumption that it was the latter, but resolving to an actual hash would be consistent with our behavior elsewhere, so I'll update.

**Pull request merged**